zigpy/zcl/clusters/closures.py

```diff
@@ -71,62 +71,62 @@ class DoorLock(Cluster):
-        0x0007: ('clear_pin_code_response', (), True),
-        0x0008: ('clear_all_pin_codes_response', (), True),
-        0x0009: ('set_user_status_response', (), True),
-        0x000a: ('get_user_status_response', (), True),
-        0x000b: ('set_week_day_schedule_response', (), True),
-        0x000c: ('get_week_day_schedule_response', (), True),
-        0x000d: ('clear_week_day_schedule_response', (), True),
-        0x000e: ('set_year_day_schedule_response', (), True),
-        0x000f: ('get_year_day_schedule_response', (), True),
-        0x0010: ('clear_year_day_schedule_response', (), True),
-        0x0011: ('set_holiday_schedule_response', (), True),
-        0x0012: ('get_holiday_schedule_response', (), True),
-        0x0013: ('clear_holiday_schedule_response', (), True),
-        0x0014: ('set_user_type_response', (), True),
-        0x0015: ('get_user_type_response', (), True),
-        0x0016: ('set_rfid_code_response', (), True),
-        0x0017: ('get_rfid_code_response', (), True),
-        0x0018: ('clear_rfid_code_response', (), True),
-        0x0019: ('clear_all_rfid_codes_response', (), True),
-        0x0020: ('operation_event_notification', (), False),
-        0x0021: ('programming_event_notification', (), False),
+        0x0000: ('lock_door_response', (foundation.Status, ), True),
+        0x0001: ('unlock_door_response', (foundation.Status, ), True),
+        0x0002: ('toggle_door_response', (foundation.Status, ), True),
+        0x0003: ('unlock_with_timeout_response', (foundation.Status, ), True),
+        0x0004: ('get_log_record_response', (t.uint16_t, t.uint32_t, t.enum8, t.uint8_t, t.uint8_t, t.uint16_t), True),
+        0x0005: ('set_pin_code_response', (foundation.Status, ), True),
+        0x0006: ('get_pin_code_response', (t.uint16_t, t.uint8_t, t.enum8, t.CharacterString), True),
+        0x0007: ('clear_pin_code_response', (foundation.Status, ), True),
+        0x0008: ('clear_all_pin_codes_response', (foundation.Status, ), True),
+        0x0009: ('set_user_status_response', (foundation.Status, ), True),
+        0x000a: ('get_user_status_response', (t.uint16_t, t.uint8_t), True),
+        0x000b: ('set_week_day_schedule_response', (foundation.Status, ), True),
+        0x000c: ('get_week_day_schedule_response', (t.uint8_t, t.uint16_t, t.uint8_t, t.uint8_t, t.uint8_t, t.uint8_t, t.uint8_t, t.uint8_t), True),
+        0x000d: ('clear_week_day_schedule_response', (foundation.Status, ), True),
+        0x000e: ('set_year_day_schedule_response', (foundation.Status, ), True),
+        0x000f: ('get_year_day_schedule_response', (t.uint8_t, t.uint16_t, foundation.Status, t.uint32_t, t.uint32_t), True),
+        0x0010: ('clear_year_day_schedule_response', (foundation.Status, ), True),
+        0x0011: ('set_holiday_schedule_response', (foundation.Status, ), True),
+        0x0012: ('get_holiday_schedule_response', (t.uint8_t, foundation.Status, t.uint32_t, t.uint32_t, t.enum8), True),
```

Comment:
last three should be optional?
![image](https://user-images.githubusercontent.com/5826160/58445720-b1ba3e00-80cb-11e9-81e7-e3933cc529f1.png)